src/org/wojcieszko/chess/Chess.java

```diff
@@ -0,0 +1,34 @@
+package org.wojcieszko.chess;
+
+import java.util.ArrayList;
+
+public class Chess {
+
+    public Tower tower = new Tower();
+
+    public ArrayList<int[]> arrayList = new ArrayList<>();
+
+    public void towerMoves(int[] position) {
+
+        int[] currentPosition;
```

Comment:
Drobiazg:
deklaracja powinna być razem z inicjalizacją, czyli w linii 17.

Takie wyniesienie poza pętlę ma sens, jeśli w zmiennej coś akumulujesz, co zbierasz w pętli i chcesz, żeby to coś było dostępne po skończeniu pętli.